src/main/java/com/microsoft/graph/requests/ResponseBodyHandler.java

```diff
@@ -0,0 +1,117 @@
+    @Override
+    @SuppressFBWarnings
+    public <NativeResponseType, ModelType> CompletableFuture<ModelType> handleResponseAsync(@Nonnull NativeResponseType response, @Nullable HashMap<String, ParsableFactory<? extends Parsable>> errorMappings) {
+        if(response instanceof Response && ((Response) response).body()!=null) {
+            ResponseBody body = ((Response) response).body();
+            try{boolean successfulResponse = validateSuccessfulResponse((Response) response, errorMappings);
+                if(successfulResponse) {
+                    return handleSuccessfulResponse(body);
+                }
+            }
+            catch(ApiException ex) {
+                CompletableFuture<ModelType> exceptionalResult = new CompletableFuture<>();
+                exceptionalResult.completeExceptionally(ex);
+                return exceptionalResult;
+            }
+        } else {
+            throw new IllegalArgumentException("The provided response type is not supported by this response handler.");
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+    @SuppressFBWarnings
+    private <ModelType> CompletableFuture<ModelType> handleSuccessfulResponse(ResponseBody body) {
+        try(final InputStream in = body.byteStream()) {
+            String[] contentType = body.contentType().toString().split(";"); //contentType.toString() returns in format <mediaType>;<charset>, we only want the mediaType.
+            byte[] responseStream = ByteStreams.toByteArray(in);
+            ParseNode jsonParseNode = this.parseNodeFactory.getParseNode(contentType[0], new ByteArrayInputStream(responseStream));
+            final ModelType result = (ModelType) jsonParseNode.getObjectValue(this.factory); //We can be sure this is the correct type since return of this method is based on the type of the factory.
+            return CompletableFuture.completedFuture(result);
+        } catch (IOException ex) {
+            CompletableFuture<ModelType> exceptionalResult = new CompletableFuture<>();
+            exceptionalResult.completeExceptionally(ex);
+            return exceptionalResult;
+        }
+    }
+    @SuppressFBWarnings
+    private boolean validateSuccessfulResponse(Response responseMessage, HashMap<String, ParsableFactory<? extends Parsable>> errorMapping) throws ApiException {
+        if (responseMessage.isSuccessful()) {
+            return true;
+        }
+        int statusCode = responseMessage.code();
+        String statusCodeString = String.valueOf(statusCode);
+        try (final InputStream in = Objects.requireNonNull(responseMessage.body()).byteStream()) {
+            String[] contentType = responseMessage.body().contentType().toString().split(";"); //contentType.toString() returns in format <mediaType>;<charset>, we only want the mediaType.
+            byte[] responseStream = ByteStreams.toByteArray(in);
+            String rawResponseBody = new String(responseStream, StandardCharsets.UTF_8);
```

Comment:
duplicated with previous method

Reply:
+1 
This should probably only check the status code and return true/false and simplify the calling code.